### PR TITLE
Updated crate to latest versions

### PR DESCRIPTION
**Pull request by aggstam**
This commit includes: updated rust edition, fixed all linting/clippy stuff based on latest standards, updated all dependencies to latest versions.
Code correctness was tested (cargo build/check/clippy/test) using --release --all-features flags.

### CODE REVIEW

**Comment by aggstam** [comment]
Hey @DemiMarie ,
Let me know if you are planning to merge this PR.
Since newer crate versions exists I can update them again.

File: src/lib.rs [hunk 1]
```diff
@@ -364,7 +363,11 @@ fn parse_input<'a>(
     const CONTEXT_SPECIFIC_PRIMITIVE_2: u8 = der::CONTEXT_SPECIFIC | 2;
     const CONTEXT_SPECIFIC_CONSTRUCTED_3: u8 = der::Tag::ContextSpecificConstructed3 as _;
     #[cfg(not(feature = "legacy-certificates"))]
-    if input.read_bytes(5).map_err(|_| Error::BadDER)? != untrusted::Input::from(&[160, 3, 2, 1, 2])
+    if input
+        .read_bytes(5)
+        .map_err(|_| Error::BadDer)?
+        .as_slice_less_safe()
+        != untrusted::Input::from(&[160, 3, 2, 1, 2]).as_slice_less_safe()
```
Comment (DemiMarie) [inline src/lib.rs:370]:
```suggestion
        != &[160, 3, 2, 1, 2])
```

**Review by DemiMarie — APPROVED** [review]
Trivial nits, thanks!

**Comment by DemiMarie** [comment]
> Hey @DemiMarie , Let me know if you are planning to merge this PR. Since newer crate versions exists I can update them again.

I forgot, thanks for the reminder!

**Comment by DemiMarie** [comment]
> Hey @DemiMarie , Let me know if you are planning to merge this PR. Since newer crate versions exists I can update them again.

That would be great!  Just curious: where are you using this code?

**Comment by aggstam** [comment]
@DemiMarie No worries, I also forgot I had this open lol.
Ok will create a new PR to update the versions again.
I'm developing stuff for RISC-V so I'm trying to "upgrade" the lower level rust crates that act as dependencies on common crates for x509 etc., so we can start having official support for that architecture in the rust ecosystem.

**Pull request merged**